### DISCUSSION TITLE
Nested/Chained Wireguard Setup

### DISCUSSION THREAD

**Discussion by asaddon** [qdm12/gluetun#1323] [Q&A]
Hello all, Happy New Year!!
I'm new to wire guard and Gluetun and I'm in a bit of dilemma. 
I own two Oracle servers one in USA and one in Dubai, I'm from Pakistan myself.
What I want to accomplish is, connect from Pakistan to my server in Dubai and from Dubai connect to the server in USA so that I can browse the internet like I'm in USA.

I'm running Portainer on Both servers, On USA Server I used wg-easy docker image to setup the Wireguard and I'm connecting to it via Gluetun custom config mode running on Dubai server, it's connecting fine but how can I create a configuration file which I use on my local machine in Pakistan to connect to Dubai which will pass traffic to USA over Gluetun.

I hope I've made it clear.

My Device in Pakistan ---> Oracle Server in Dubai ---> Oracle Server in USA
I prefer to do all this in Docker via Portainer.

**Comment by bnhf** [comment]
@asaddon 

Interesting project. Given what you have up-and-running already, how about using the built-in Gluetun Shadowsocks proxy for the Pakistan to Dubai link? You'd have password protection and encryption for the Shadowsocks port you'd open on your router in Dubai.

You'd need a Shadowsocks client in Pakistan, which could be run on a per PC basis, directly on some routers, or in a container. Doing it this way keeps the setup in Dubai fairly straightforward.

**Reply by asaddon** [reply]
Yup I know about the Shadowsocks method and I will eventually do it that way. I'm just trying to learn Wireguard and looking for a way to do it Gluetun way (or even natively in Docker)
